core/rpc-client/src/direct_client.rs

```diff
@@ -32,15 +35,46 @@ impl Handler for WsClient {
 			Err(e) => Err(e),
 		}
 	}
+
 	fn on_message(&mut self, msg: Message) -> ClientResult<()> {
 		info!("got message");
 		debug!("{}", msg);
+		info!("sending result to MpscSender..");
 		self.result.send(msg.to_string()).unwrap();
 		if !self.do_watch {
+			info!("do_watch is false, closing connection");
 			self.out.close(CloseCode::Normal).unwrap();
+			info!("connection is closed");
 		}
+		info!("on_message successful, returning");
 		Ok(())
 	}
+
+	/// we are overriding the `upgrade_ssl_client` method in order to disable hostname verification
+	/// this is taken from https://github.com/housleyjk/ws-rs/blob/master/examples/unsafe-ssl-client.rs
+	/// TODO: hostname verification should probably be enabled again for production?
+	fn upgrade_ssl_client(
+		&mut self,
+		sock: TcpStream,
+		_: &url::Url,
+	) -> ws::Result<SslStream<TcpStream>> {
+		let mut builder = SslConnector::builder(SslMethod::tls_client()).map_err(|e| {
+			ws::Error::new(
+				ws::ErrorKind::Internal,
+				format!("Failed to upgrade client to SSL: {}", e),
+			)
+		})?;
+		builder.set_verify(SslVerifyMode::empty());
+
+		let connector = builder.build();
+		connector
+			.configure()
+			.unwrap()
+			.use_server_name_indication(false)
+			.verify_hostname(false)
+			.connect("", sock)
+			.map_err(From::from)
+	}
```

Comment:
Override the `upgrade_ssl_client` method to enable TLS. Also had to disable some checks (hostname, SNI), which should be temporary.